pkg/assembler/helpers/purl.go

```diff
@@ -37,80 +40,109 @@ func PurlToPkg(purlUri string) (*model.Package, error) {
 	return purlConvert(p)
 }
 
+// purlConvert converts a purl URI into a graphql package node.
 func purlConvert(p purl.PackageURL) (*model.Package, error) {
-	switch p.Type {
-
 	// Enumeration of https://github.com/package-url/purl-spec#known-purl-types
 	// TODO(lumjjb): each PURL definition usually comes with a default repository
 	// we should consider addition of default repository to the prefix of the namespace
 	// so that they can be referenced with higher specificity in GUAC
 	//
 	// PURL types not defined in purl library handled generically
```

Comment:
this comment is for the text defined ones in the case statements (so would be comments for the rows in the map below, and then another comment for PurlTypeGUAC case and 1 more for the rest that are from PURL library